### PR TITLE
feat: add tiproxy tiup pipeline

### PR DESCRIPTION
**Pull request by lijie0123**
## Why:
- tiproxy need to publish to tiup, with cd golang builder

### CODE REVIEW

File: jenkins/jobs/cd/release_tiproxy_tiup.groovy [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                            timeout(5)
+                        }
+                    }
+                }
+            }
+        }
+    }
+    parameters {
+        parameters{
+            stringParam('GitRef', 'v0.1.1', 'tiproxy repo git reference')
```
Comment (wuhuizuo) [inline jenkins/jobs/cd/release_tiproxy_tiup.groovy:25]: Why tag, branch is better.

File: jenkins/pipelines/cd/release-tiproxy-tiup.groovy [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    sh "make cmd"
+}
+
+pipeline{
+    environment {
+        GOPROXY = "http://goproxy.pingcap.net,https://proxy.golang.org,direct"
+    }
+    agent none
+    stages{
+        stage("build"){
```
Comment (purelind) [inline jenkins/pipelines/cd/release-tiproxy-tiup.groovy:18]:
```suggestion
        stage("build") {
```

**Pull request merged**